dap_aria_mapping/pipeline/taxonomy_development/community_detection.py

```diff
@@ -0,0 +1,168 @@
+                node_next_level_community = node_upper_level_community + "_{}".format(
+                    str(index)
+                )
+                hierarchy[node] = node_next_level_community
+            hierarchy = generate_hierarchy(
+                node_list,
+                network,
+                hierarchy,
+                resolution + resolution_increments,
+                n_iter + 1,
+            )
+    return hierarchy
+
+
+def format_output(hierarchy: dict, total_splits: int) -> pd.DataFrame:
+    """formats the clustering hierarchy into table
+
+    Args:
+        hierarchy (dict): clustering hierarchy
+        total_splits (int): total number of levels in the hierarchy
+
+    Returns:
+        pd.DataFrame: table where index is entity name and columns are cluster assignments at each level
+    """
+    # parse community string and format as list
+    output_list = []
+    for entity, community in hierarchy.items():
+        temp = [None] * (total_splits + 1)
+        temp[0] = entity
+        split_communs = community.split("_")
+        for i in range(1, total_splits + 1):
+            commun_str = "_".join(split_communs[:i])
+            try:
+                temp[i] = commun_str
+            except IndexError:
+                print("Error with entity {} in community {}".format(entity, community))
+                continue
+        output_list.append(temp)
+
+    # build dataframe and set column names and index
+    output_columns = ["Entity"]
+    for i in range(1, total_splits + 1):
+        output_columns.append("Level_{}".format(i))
+    output_df = pd.DataFrame(output_list)
+    output_df.columns = output_columns
+    output_df.set_index("Entity", inplace=True)
+
+    return output_df
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        description="Test Mode", formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "-t",
+        "--test_mode",
+        action="store_true",
+        help="run script in test mode on small sample",
+    )
+    args = parser.parse_args()
+
+    # load pre-computed cooccurrence network from S3
+    print("Loading network")
+    if args.test_mode:
+        network = get_test_cooccurrence_network()
+    else:
+        network = get_cooccurrence_network()
+
+    # load taxonomy config file with parameters for community detection
+    config = get_taxonomy_config()["community_detection"]
+
+    # recursively split network into communities
+    print("Splitting network into communities")
+    starting_resolution = config["starting_resolution"]
+    initial_partitions = louvain_communities(
+        network, resolution=starting_resolution, seed=1, weight="association_strength"
+    )
+    numbered_communities = enumerate(initial_partitions)
+
+    hierarchy = defaultdict(str)
+    for index, node_list in numbered_communities:
+        for node in node_list:
+            hierarchy[node] = str(index)
```

Comment:
Could you generate the first hierarchy by using the recursive function but and adding a couple of conditional statements inside it to modify the behaviour on the first iteration?